### PR TITLE
feat: add monitoring CRD check to database and storage webhooks

### PR DESCRIPTION
**Pull request by zeruk**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #140 

## What is the new behavior?

-
-
-

### CODE REVIEW

File: api/v1alpha1/isMonitoringCRDExists.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package v1alpha1
+
+import (
+	kube "k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func isMonitoringCRDExists(manager ctrl.Manager) bool {
```
Comment (Jorres) [inline api/v1alpha1/isMonitoringCRDExists.go:8]: Let's go with `isMonitoringCRDPresent`

File: api/v1alpha1/database_webhook.go [hunk 2]
```diff
@@ -123,6 +126,12 @@ func (r *Database) ValidateCreate() error {
 		}
 	}
 
+	if r.Spec.Monitoring != nil {
+		if !isMonitoringCRDExists(manager) {
+			return fmt.Errorf("Monitoring CRDs (`monitoring.coreos.com/v1/servicemonitor`) do not exists, please check if prometheus CRDs installed correctly")
```
Comment (Jorres) [inline api/v1alpha1/database_webhook.go:131]: Let's make this error the same as the error in line 153?

File: api/v1alpha1/isMonitoringCRDExists.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	config := manager.GetConfig()
+	clientset, err := kube.NewForConfig(config)
+	if err != nil {
+		databaselog.Error(err, "unable to get clientset while checking monitoring CRD")
+		panic(err)
```
Comment (Jorres) [inline api/v1alpha1/isMonitoringCRDExists.go:13]: The operator must not `panic` here. Imagine the situation: cluster running for ages, and then the guy accidentally enables monitoring in the middle of the update, then in `ValidateUpdate` this function is called and boom, the operator is down, everyone cries)

File: api/v1alpha1/storage_webhook.go [hunk 4]
```diff
@@ -112,13 +112,25 @@ func (r *Storage) ValidateCreate() error {
 		}
 	}
 
+	if r.Spec.Monitoring != nil {
+		if !isMonitoringCRDExists(manager) {
+			return fmt.Errorf("Monitoring CRDs (`monitoring.coreos.com/v1/servicemonitor`) do not exists, please check if prometheus CRDs installed correctly")
```
Comment (Jorres) [inline api/v1alpha1/storage_webhook.go:117]: The same issue here, please make the error message consistent. Maybe even make this whole new block (115-120) a function, to not duplicate it four times?

File: docs/tests.md [hunk 5]
```diff
@@ -56,7 +56,7 @@ Run the following (from the root of the repository):
 
 ```
 make envtest
-./bin/setup-envtest 1.26
+./bin/setup-envtest use 1.26
```
Comment (Jorres) [inline docs/tests.md:59]: Can you please explain what does this change do?

File: api/v1alpha1/isMonitoringCRDExists.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	kube "k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func isMonitoringCRDExists(manager ctrl.Manager) bool {
+	config := manager.GetConfig()
+	clientset, err := kube.NewForConfig(config)
+	if err != nil {
+		databaselog.Error(err, "unable to get clientset while checking monitoring CRD")
```
Comment (Jorres) [inline api/v1alpha1/isMonitoringCRDExists.go:12]: This is wrong, this error may happen on Storage or Database, but the error message will always be written into a database log. Maybe we can pass the log as a variable into this function?

**Review by Jorres — CHANGES_REQUESTED** [review]

File: docs/tests.md (hunk 5 above)
Comment (zeruk) [reply docs/tests.md:59]: it just corrects wrong usage of application

**Review by Jorres — APPROVED** [review]

**Pull request merged**